src/ui/public/modals/__tests__/confirm_modal.js

```diff
@@ -0,0 +1,186 @@
+    });
+
+    it('when showClose is on but title is not given', function () {
+      const options = { customConfirmButton: 'b', onConfirm: _.noop, showClose: true };
+      expect(() => confirmModal('hi', options)).to.throwError();
+    });
+  });
+
+  it('shows the message', function () {
+    const myMessage = 'Hi, how are you?';
+    confirmModal(myMessage, { confirmButtonText: 'GREAT!', onConfirm: _.noop });
+
+    $rootScope.$digest();
+    const message = findByDataTestSubj('confirmModalBodyText')[0].innerText;
+    expect(message).to.equal(myMessage);
+  });
+
+  describe('shows custom text', function () {
+    const confirmModalOptions = {
+      confirmButtonText: 'Troodon',
+      cancelButtonText: 'Dilophosaurus',
+      title: 'Dinosaurs',
+      onConfirm: _.noop
+    };
+
+    it('for confirm button', () => {
+      confirmModal('What\'s your favorite dinosaur?', confirmModalOptions);
+      $rootScope.$digest();
+      const confirmButtonText = findByDataTestSubj('confirmModalConfirmButton')[0].innerText;
+      expect(confirmButtonText).to.equal('Troodon');
+    });
+
+    it('for cancel button', () => {
+      confirmModal('What\'s your favorite dinosaur?', confirmModalOptions);
+      $rootScope.$digest();
+      const cancelButtonText = findByDataTestSubj('confirmModalCancelButton')[0].innerText;
+      expect(cancelButtonText).to.equal('Dilophosaurus');
+    });
+
+    it('for title text', () => {
+      confirmModal('What\'s your favorite dinosaur?', confirmModalOptions);
+      $rootScope.$digest();
+      const titleText = findByDataTestSubj('confirmModalTitleText')[0].innerText;
+      expect(titleText).to.equal('Dinosaurs');
+    });
+  });
+
+  describe('x icon', function () {
+    it('is visible when showClose is true', function () {
+      const confirmModalOptions = {
+        confirmButtonText: 'bye',
+        onConfirm: _.noop,
+        showClose: true,
+        title: 'hi'
+      };
+      confirmModal('hi', confirmModalOptions);
+
+      $rootScope.$digest();
+      const xIcon = findByDataTestSubj('confirmModalCloseButton');
+      expect(xIcon.length).to.be(1);
+    });
+
+    it('is not visible when showClose is false', function () {
+      const confirmModalOptions = {
+        confirmButtonText: 'bye',
+        onConfirm: _.noop,
+        title: 'hi',
+        showClose: false
+      };
+      confirmModal('hi', confirmModalOptions);
+
+      $rootScope.$digest();
+      const xIcon = findByDataTestSubj('confirmModalCloseButton');
+      expect(xIcon.length).to.be(0);
+    });
+  });
+
+  describe('callbacks are called:', function () {
+    const confirmCallback = sinon.spy();
+    const closeCallback = sinon.spy();
+    const cancelCallback = sinon.spy();
+
+    const confirmModalOptions = {
+      confirmButtonText: 'bye',
+      onConfirm: confirmCallback,
+      onCancel: cancelCallback,
+      onClose: closeCallback,
```

Comment:
Yeah, we can punt on it for now. I think it's enough that onClose defaults to onCancel, since it doesn't change the way this dialog works, and it should no longer trip up a developer trying to use it.